packages/dds/tree/src/feature-libraries/schemaIndex.ts

```diff
@@ -105,10 +243,32 @@ export class SchemaIndex implements Index<unknown>, SummaryElement {
     }
 
     public async load(services: IChannelStorageService, parse: SummaryElementParser): Promise<void> {
-        if (await services.contains(schemaBlobKey)) {
-            const blob = await services.readBlob(schemaBlobKey);
-            parse(bufferToString(blob, "utf-8")) as string;
-            // TODO: use parsed results to rehydrate the schema
+        const [hasString, hasBlob] = await Promise.all(
+            [services.contains(schemaStringKey), services.contains(schemaBlobKey)]);
+        assert(hasString || hasBlob, "Schema is required in summary");
+        let schemaBuffer: ArrayBufferLike;
+        if (hasBlob) {
+            const handleBuffer = await services.readBlob(schemaBlobKey);
+            const handleString = bufferToString(handleBuffer, "utf-8");
+            const handle = parse(handleString) as IFluidHandle<ArrayBufferLike>;
+            schemaBuffer = await handle.get();
+        } else {
+            schemaBuffer = await services.readBlob(schemaStringKey);
+        }
+
+        assert(this.schema.treeSchema.size === 0,
+            "there should not already be stored schema when loading stored schema");
+        assert(this.schema.globalFieldSchema.size === 0,
+            "there should not already be stored schema when loading stored schema");
```

Comment:
nit: move these to the top of the method

Reply:
I specifically put these after the awaits so if concurrent code adds schema, and we stomp on it its still an error.
```suggestion
        // After the awaits, validate that the schema is in a clean state.
        // This detects any schema that could have been accidently added through
        // invalid means and are about to be overwritten.
        assert(this.schema.treeSchema.size === 0,
            "there should not already be stored schema when loading stored schema");
        assert(this.schema.globalFieldSchema.size === 0,
            "there should not already be stored schema when loading stored schema");
```

Reply:
Ah I see, good thinking. Otherwise it would silently overwrite the other schema.